windows/src/desktop/kmshell/install/UfrmInstallKeyboard.pas

```diff
@@ -161,9 +185,29 @@ function InstallFiles(Owner: TComponent; const FileNames: TWideStrings; ASilent:
     try
       kmcom.Keyboards.Refresh;
 
-      if AnsiSameText(ExtractFileExt(FileNames[i]), '.kmp')
-        then kmcom.Packages.Install(FileNames[i], True)
-        else kmcom.Keyboards.Install(FileNames[i], True);
+      FilenameBCP47 := Filenames[I].Split(['=']);
+      Filename := FilenameBCP47[0];
+      IsPackage := AnsiSameText(ExtractFileExt(FileNames[i]), '.kmp');
+
+      if (Length(FilenameBCP47) > 1) and (FilenameBCP47[1] <> '') then
+      begin
+        if IsPackage then
+        begin
+          FPackage := (kmcom.Packages as IKeymanPackagesInstalled2).Install2(FileNames[i], True, False);
+          InstallKeyboardPackageLanguage(FPackage, FilenameBCP47[2]);
```

Comment:
Using your example of `keyman-setup.khmer_angkor.km.exe` as a filename, what do we expect to see here?   I'm trying to compare it against your example (`keyman-setup.khmer_angkor.km.exe`), but I'm coming up short.  (In particular, because of the `Filenames[I].Split(['=']);` line above; it's not clear where a `=` comes into play at all.

Reply:
So this is parsing a command-line parameter for kmshell.exe, not the bootstrap. This uses the pattern `-i package_filename[=bcp47] [package_filename[=bcp47]] ...`

For example: `kmshell -i c:\temp\khmer_angkor.kmp=km c:\temp\sil_euro_latin.kmp=fr`

The bootstrap installer will parse its filename (which cannot safely use `=`, hence the `.` strategy), download the .kmp files, and pass the parameters into Keyman Configuration (kmshell).

In some ways it'd be nicer for the two strategies to be more consistent in their patterns, but I don't want to constrain the parameter model to the limited character set of the filename model...

(Note that the bootstrap installer *also* accepts a `-p package_id=bcp47[,package_id=bcp47]` parameter, which works the same way as embedding into the bootstrap filename but is more nicerer for running from command line).

Reply:
Thanks, that's the missing information I was looking for.  Makes a lot more sense now.  Might be useful to have this in comments / in-code documentation somewhere.